src/app/ReadClient.cpp

```diff
@@ -493,6 +494,18 @@ void ReadClient::OnActiveModeNotification()
         return;
     }
 
+    // If the server sends out check-in message, and there is no reschedule subscription yet in client side at the same time, it means
+    // current client does not realize subscription has gone, and we should forcibly timeout current subscription, and schedule a new one.
```

Comment:
So the thing is: per spec, the check-in target and the thing whose subscription the server checks for might be different nodes.  And in practice this is actually going to happen, I can guarantee that.

So it's entirely possible for this subscription to be _just fine_ and yet for a check-in message to arrive because some _other_ subscription from a different node is not fine.

I assume we're already surfacing check-in messages to the SDK API consumer?  If so, they would need to retrigger subscriptions as needed.  Or we need some way to configure this behavior, at the very least, because the right thing to do here depends on information this code does not have.

Reply:
yes, we have already surfacing check-in messages to the SDK API consumer, it is consumer's reponsibility to retrigger subscription as need, they can mimic OnActiveModeNotification behavior to configure their behavior.

Reply:
> So the thing is: per spec, the check-in target and the thing whose subscription the server checks for might be different nodes. And in practice this is actually going to happen, I can guarantee that.

If subscription from node A is and good, register node for check-in is for nodeB,  check-in message would never arrive at nodeA, right? since server send check-in message based upon registered node, unless we allow that multiple nodes reside in same client device? maybe we need add one more check between ICDClientInfo::check_in_node against the node id extracted from readClient in InteractionModelEngine:: OnActiveModeNotification?

Reply:
just updated code to reflect the above change.

Reply:
> If subscription from node A is and good, register node for check-in is for nodeB, check-in message would never arrive at nodeA, right?

Wrong scenario.  The scenario is that node A is check-in target and has a subscription but that is NOT the subscription check-in is tracking.  Node B is the one the check-in is tracking and that one is dead.  Node A will receive the check-in, because Node B is not subscribed, and the right action is to prod Node B (or maybe Node C, depending) in the app, not tear down A's perfectly OK subscription.

Point being: ReadClient does not have the information to make this decision; the application does.